### PR TITLE
Course grading page

### PR DESCRIPTION
**Pull request by proinators**
## Description

<pre>
Introducing a Course grading page, where midsem grades from ERP can be uploaded as images to the website for each course.
</pre>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New website feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New script (new python script for handouts handling)
- [ ] New resource (new file that is added)
  - [ ] Adding new / updating notes
  - [ ] Adding new / updating handouts
  - [ ] Adding new / updating chronicles
  - [ ] Adding new / updating reviews
- [ ] Misc (explain in description)

## Checklist:

- [x] I used `pnpm` for my project dependencies
- [x] I didnt push any `.env` files

### CODE REVIEW

File: pages/courses/grading/index.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import { useSession } from "next-auth/react";
+import { PreReqGroup } from "@/types/PreReq";
+import Modal from "@/components/Modal";
+import { toast } from 'react-toastify';
+import { CourseGrading } from "@/types/CourseGrading";
+
+export const getStaticProps: GetStaticProps = async () => {
+    const fs = require("fs");
+
+    let prereqs = fs.readFileSync("./public/prereqs.json").toString();
```
Comment (thenicekat) [inline pages/courses/grading/index.tsx:14]: huh?

File: types/CourseGrading.tsx
Comment (thenicekat) [inline types/CourseGrading.tsx:1]: New line at EOF

File: pages/api/courses/grading/get.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            error: true,
+            data: []
+        })
+        return;
+    }
+
+    const email = session?.user?.email
+    if (!email?.includes('hyderabad')) {
+        res.status(403).json({
+            message: 'Course reviews are only available for BITS Hyderabad students',
```
Comment (thenicekat) [inline pages/api/courses/grading/get.ts:30]:
```suggestion
            message: 'Midsem grading feature is only available for BITS Hyderabad students',
```

**Review by thenicekat — CHANGES_REQUESTED** [review]

File: pages/api/courses/grading/get.ts (hunk 2 above)
Comment (proinators) [reply pages/api/courses/grading/get.ts]: Done

File: pages/api/courses/grading/add.ts
Comment (thenicekat) [inline pages/api/courses/grading/add.ts:1]: EOF.

File: pages/courses/grading/add.tsx
Comment (thenicekat) [inline pages/courses/grading/add.tsx:1]: Just verify edge cases.

**Review by thenicekat — APPROVED** [review]

File: pages/api/courses/grading/add.ts
Comment (proinators) [reply pages/api/courses/grading/add.ts:1]: Done

**Pull request merged**